src/Interfaces/delivery/UndeliverableDate.ts

```diff
@@ -0,0 +1,3 @@
+export default class UndeliverableDate {
+    date!: Date;
```

Comment:
Should this have an exclamation mark? It's telling the compiler that, even if it thinks this might be undefined, to ignore that as there's no possible way.

I don't think we'd want to use it in the context of an interface like this


I'm not even certain it is doing anything here, as `date?: Date` would be Date or undefined, and `date: Date | null` would be Date or null. So just `date: Date` should mean this interface always has a date property and it is always a Date.

```suggestion
    date: Date;
```

Reply:
I agree with this however I get this error 

`Property 'date' has no initializer and is not definitely assigned in the constructor.`

Reply:
In which file and what line?

Reply:
src/Interfaces/delivery/UndeliverableDate.ts:2:5 - error TS2564: Property 'date' has no initializer and is not definitely assigned in the constructor.

Not sure if its my

Reply:
Ah, you've made it a class. This is supposed to be an interface right?

Reply:
ahh yeah that's the one thanks